flex/engines/graph_db/grin/src/predefine.h

```diff
@@ -0,0 +1,66 @@
+#include "grin/predefine.h"
+
+#include "storages/rt_mutable_graph/mutable_property_fragment.h"
+
+typedef gs::oid_t GRIN_OID_T;
+typedef gs::vid_t GRIN_VID_T;
+
+typedef gs::MutablePropertyFragment GRIN_GRAPH_T;
+
+typedef struct GRIN_VERTEX_T {
+  uint8_t label;
+  uint32_t vid;
+} GRIN_VERTEX_T;
+
+typedef struct GRIN_EDGE_T {
+  GRIN_VERTEX_T dst;
+  GRIN_VERTEX_T src;
+  GRIN_DIRECTION dir;
+  gs::label_t label;
+  gs::Any data;
+} GRIN_EDGE_T;
+
+#ifdef GRIN_ENABLE_ADJACENT_LIST
+typedef struct GRIN_ADJACENT_LIST_T {
+  GRIN_VERTEX_T v;
+  GRIN_DIRECTION dir;
+  GRIN_EDGE_TYPE edge_label;
+} GRIN_ADJACENT_LIST_T;
+#endif
+
+#ifdef GRIN_ENABLE_ADJACENT_LIST_ITERATOR
+typedef struct GRIN_ADJACENT_LIST_ITERATOR_T {
+  std::shared_ptr<gs::MutableCsrConstEdgeIterBase> edge_iter;
```

Comment:
copying shared_ptr is a nightmare for efficiency, please be careful and make sure this is no unnecessary copying happens.